src/curl.cpp

```diff
@@ -346,7 +346,7 @@ long             S3fsCurl::connect_timeout     = 300;  // default
 time_t           S3fsCurl::readwrite_timeout   = 60;   // default
 int              S3fsCurl::retries             = 3;    // default
 bool             S3fsCurl::is_public_bucket    = false;
-string           S3fsCurl::default_acl         = "private";
+string           S3fsCurl::default_acl         = "";
```

Comment:
@andrewgaul @sqlbot Thanks for your help.

If there is a possibility that the behavior of existing s3fs users will be changed, I think that it is better not to change the default value.
This is why s3fs collaborators including us do not make it hard to support about this(if there is an issue).
By merging your patches without changing default value, s3fs can provide a way to not send x-amz-acl and users can control the sending of this headers as needed.
I think that it will be able to deal with StorageGRID etc even without changing the default value.
(In order not to send this header, I think that it is possible by specifying option 'default_acl=""'. If necessary, I think that we may make no_acl_header with the same meaning as it.)

I'd like to get your thoughts on it.